File: dkplayer-players/ijk/src/main/java/xyz/doikki/videoplayer/ijk/IjkPlayerFactory.java
package xyz.doikki.videoplayer.ijk;

import android.content.Context;

import xyz.doikki.videoplayer.player.PlayerFactory;

public class IjkPlayerFactory implements PlayerFactory {

    public static IjkPlayerFactory create() {
        return new IjkPlayerFactory();
    }

    @Override
    public IjkDKPlayer create(Context context) {
        return new IjkDKPlayer(context);
    }
}

File: dkplayer-players/ijk/src/main/java/xyz/doikki/videoplayer/ijk/IjkPlayerFactory.kt
package xyz.doikki.videoplayer.ijk

import android.content.Context
import xyz.doikki.videoplayer.player.PlayerFactory

class IjkPlayerFactory : PlayerFactory {
    override fun create(context: Context): IjkPlayer {
        return IjkPlayer(context)
    }

    companion object {
        @JvmStatic
        fun create(): IjkPlayerFactory {
            return IjkPlayerFactory()
        }
    }
}